src/stan/lang/grammars/semantic_actions_def.cpp

```diff
@@ -1844,11 +1899,35 @@ namespace stan {
 
       fun.type_ = function_signatures::instance()
         .get_result_type(fun.name_, arg_types, error_msgs);
+
       if (fun.type_ == ILL_FORMED_T) {
         pass = false;
         return;
       }
 
+      // get function definition for this functiion
+      std::vector<function_arg_type> fun_arg_types;
+      for (size_t i = 0; i < fun.args_.size(); ++i)
+        fun_arg_types.push_back(function_arg_type(arg_types[i]));
+      function_signature_t sig(fun.type_, fun_arg_types);
+      function_signature_t decl_sig =
+        function_signatures::instance().get_definition(fun.name_, sig);
+      if (!decl_sig.first.is_ill_formed()) {
+        for (size_t i = 0; i < fun_arg_types.size(); ++i) {
+          if (decl_sig.second[i].data_only_) {
```

Comment:
Instead of 

```
if (cond1)
  if (cond2) 
    body;
```

you can lower nesting with

```
if (cond1 && cond2)
  body;
```

Reply:
fixed.